engine/src/main/java/org/pentaho/di/trans/steps/mailinput/MailInput.java

```diff
@@ -0,0 +1,632 @@
+
+        if ( data.mailConn.getSearchTerm() != null ) { // add search filter
+          data.folderIterator =
+            new SearchEnabledFolderIterator( data.folderIterator, data.mailConn.getSearchTerm() );
+        }
+      } else { // fetch all
+        data.mailConn.retrieveMessages();
+        data.folderIterator = new ArrayIterator( data.mailConn.getMessages() );
+      }
+
+      if ( isDebug() ) {
+        logDebug( BaseMessages.getString( PKG, "MailInput.Log.MessagesInFolder", data.folder, data.messagesCount ) );
+      }
+
+    } catch ( Exception e ) {
+      logError( "Error opening folder " + data.folderenr + " " + data.folder + ": " + e.toString() );
+      logError( Const.getStackTracker( e ) );
+      stopAll();
+      setErrors( 1 );
+      return false;
+    }
+    return true;
+  }
+
+  public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
+    meta = (MailInputMeta) smi;
+    data = (MailInputData) sdi;
+
+    if ( !super.init( smi, sdi ) ) {
+      return false;
+    }
+
+    if ( !meta.isDynamicFolder() ) {
+      try {
+        // Create the output row meta-data
+        data.outputRowMeta = new RowMeta();
+        meta.getFields( data.outputRowMeta, getStepname(), null, null, this, repository, metaStore ); // get the
+                                                                                                      // metadata
+                                                                                                      // populated
+
+      } catch ( Exception e ) {
+        logError( BaseMessages.getString( PKG, "MailInput.ErrorInit", e.toString() ) );
+        logError( Const.getStackTracker( e ) );
+        return false;
+      }
+    }
+    data.usePOP = meta.getProtocol().equals( MailConnectionMeta.PROTOCOL_STRING_POP3 );
+
+    String realserver = environmentSubstitute( meta.getServerName() );
+    if ( meta.getProtocol().equals( MailConnectionMeta.PROTOCOL_STRING_MBOX )
+      && StringUtils.startsWith( realserver, "file://" ) ) {
+      realserver = StringUtils.remove( realserver, "file://" );
+    }
+
+    String realusername = environmentSubstitute( meta.getUserName() );
+    String realpassword = Utils.resolvePassword( variables, meta.getPassword() );
+    int realport = Const.toInt( environmentSubstitute( meta.getPort() ), -1 );
+    String realProxyUsername = environmentSubstitute( meta.getProxyUsername() );
+    if ( !meta.isDynamicFolder() ) {
+      //Limit field has absolute priority
+      String reallimitrow = environmentSubstitute( meta.getRowLimit() );
+      int limit = Const.toInt( reallimitrow, 0 );
+      //Limit field has absolute priority
+      if ( limit == 0 ) {
+        limit = getReadFirst( meta.getProtocol() );
+      }
+      data.rowlimit = limit;
+    }
+    Date beginDate = null;
+    Date endDate = null;
+    SimpleDateFormat df = new SimpleDateFormat( MailInputMeta.DATE_PATTERN );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.